src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs

```diff
@@ -129,23 +129,17 @@ private void EnsureStartup()
 
             if (StartupType == null)
             {
-                var diagnosticTypeMessages = new List<string>();
-                StartupType = _startupLoader.FindStartupType(StartupAssemblyName, diagnosticTypeMessages);
+                StartupType = _startupLoader.FindStartupType(StartupAssemblyName);
                 if (StartupType == null)
                 {
-                    throw new ArgumentException(
-                        diagnosticTypeMessages.Aggregate("Failed to find a startup type for the web application.", (a, b) => a + "\r\n" + b),
-                        StartupAssemblyName);
+                    throw new ArgumentException("Failed to find a startup type for the web application.", StartupAssemblyName);
```

Comment:
Shouldn't this be nameof(StartupAssemblyName)? I am pretty sure the second parameter is the name of the parameter causing the issue. Currently you will get something like "Parameter Name: MyAssembly"
(This also tells me that there are no tests for this code path)